wasm/main.py

```diff
@@ -609,31 +639,34 @@ def spec_validate_start(C, start):
 # 3.4.8 EXPORTS
 
 
-def spec_validate_export(C, export):
-    return spec_validate_exportdesc(C, export["desc"])
+TExportValue = Union[FuncType, TableType, MemoryType, GlobalType]
 
 
-def spec_validate_exportdesc(C, exportdesc):
-    x = exportdesc[1]
-    if exportdesc[0] == "func":
-        if len(C["funcs"]) <= x:
+def spec_validate_export(C: Context, export: Export) -> TExportValue:
+    return spec_validate_exportdesc(C, export.desc)
+
+
+def spec_validate_exportdesc(C: Context,
+                             desc: ExportDesc) -> TExportValue:
+    if isinstance(desc, FuncIdx):
+        if len(C["funcs"]) <= desc:
             raise InvalidModule("invalid")
-        return ["func", C["funcs"][x]]
-    elif exportdesc[0] == "table":
-        if len(C["tables"]) <= x:
+        return C["funcs"][desc]
```

Comment:
Huh, interesting that the caller didn't explicitly need to have this type hint handed to them `"func"`. I guess the caller always knows what kind of value it is looking up?